source/extensions/filters/http/cache/http_cache.cc

```diff
@@ -0,0 +1,156 @@
+#include "extensions/filters/http/cache/http_cache.h"
+
+#include <algorithm>
+#include <ostream>
+
+#include "common/http/headers.h"
+#include "common/protobuf/utility.h"
+
+#include "extensions/filters/http/cache/http_cache_utils.h"
+
+#include "absl/time/time.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+std::ostream& operator<<(std::ostream& os, CacheEntryStatus status) {
+  switch (status) {
+  case CacheEntryStatus::Ok:
+    return os << "Ok";
+  case CacheEntryStatus::Unusable:
+    return os << "Unusable";
+  case CacheEntryStatus::RequiresValidation:
+    return os << "RequiresValidation";
+  case CacheEntryStatus::FoundNotModified:
+    return os << "FoundNotModified";
+  case CacheEntryStatus::UnsatisfiableRange:
+    return os << "UnsatisfiableRange";
+  }
+  NOT_REACHED_GCOVR_EXCL_LINE;
+}
+
+std::ostream& operator<<(std::ostream& os, const AdjustedByteRange& range) {
+  return os << "[" << range.begin() << "," << range.end() << ")";
+}
+
+LookupRequest::LookupRequest(const Http::HeaderMap& request_headers, SystemTime timestamp)
+    : timestamp_(timestamp),
+      request_cache_control_(request_headers.CacheControl() == nullptr
+                                 ? ""
+                                 : request_headers.CacheControl()->value().getStringView()) {
+  ASSERT(request_headers.Path(), "Can't form cache lookup key for malformed Http::HeaderMap "
+                                 "with null Path.");
```

Comment:
Path can be absent for CONNECT, so to future proof this can you explicitly handle this? cc @alyssawilk. (Our goal is to have an explicit CONNECT route matcher, but it still shouldn't crash if someone incorrectly configures the caching filter for a CONNECT request).

Reply:
These ASSERTs check LookupRequest's documented prerequisites. Creating a LookupRequest for a request that can't be looked up makes no sense and would be a bug. In the case of CONNECT requests, CacheFilter will see that the method isn't cacheable, and will just return Http::FilterHeadersStatus::Continue.

Reply:
Fair enough. Can you add a small comment that mentions that we assume that any requests that doesn't satisfy these conditions will not be looked up at all? It's not clear what this assertion means exactly.

Reply:
Done.